src/daft-local-execution/src/sinks/hash_join.rs

```diff
@@ -200,7 +205,27 @@ impl IntermediateOperator for HashJoinProber {
         let left_table = left_growable.build()?;
         let right_table = right_growable.build()?;
 
-        let final_table = left_table.union(&right_table)?;
+        let common_join_keys = self
```

Comment:
this should be the same for all morsels and should just be computed once and stored in the prober

Reply:
we should just be able to store `pruned_right_side_columns`

Reply:
Make sense, fixed